flow/scripts/open.tcl

```diff
@@ -60,12 +60,62 @@ proc read_timing {input_file} {
 if {[env_var_equals GUI_TIMING 1]} {
   puts "GUI_TIMING=1 reading timing, takes a little while for large designs..."
   read_timing $input_file
+
+  gui::save_display_controls
+
+  set height [[[ord::get_db_block] getBBox] getDY]
+  set height [ord::dbu_to_microns $height]
+  set resolution [expr $height / 1000]
+
+  # FIXME IRDrop heatmap should be added, but it has to be skipped
+  # when there is no IRDrop heatmap to be rendered.
+
+  set block [ord::get_db_block]
+  set insts [$block getInsts]
+  set placed 1
+  foreach inst $insts {
+    set status [$inst getPlacementStatus]
+    # status is not in the list of PLACED, LOCKED
+    if {[lsearch {PLACED LOCKED} $status] == -1} {
+      set placed 0
+      break
+    }
+  }
+
+  set have_routes [expr {$placed && [grt::have_routes]}]
```

Comment:
Do we still need to compute/check $placed?  I thought this was fixed.

Reply:
Yes: there is no ppl::have_placed method. I can simplify the code a bit here though for the have_routes case.